<a href="https://colab.research.google.com/github/niharika-kundur/play_store_app_review_analysis/blob/main/Copy_of_Play_Store_App_Review_Analysis_Capstone_Project.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

## <b> The Play Store apps data has enormous potential to drive app-making businesses to success. Actionable insights can be drawn for developers to work on and capture the Android market. </b>

## <b> Each app (row) has values for catergory, rating, size, and more. Another dataset contains customer reviews of the android apps.</b>

## <b> Explore and analyze the data to discover key factors responsible for app engagement and success. </b>

In [2]:
import numpy as np
import pandas as pd
from matplotlib import pyplot as plt
%matplotlib inline
import seaborn as sns
import warnings
warnings.filterwarnings('ignore')

In [3]:
from google.colab import drive
drive.mount('/content/drive')


Mounted at /content/drive


In [4]:
playstore_data_path='/content/drive/MyDrive/EDA /Copy of Play Store Data.csv'
user_reviews__path='/content/drive/MyDrive/EDA /Copy of User Reviews.csv'
playstore_data_df=pd.read_csv(playstore_data_path)
user_reviews_df=pd.read_csv(user_reviews__path)



In [5]:
playstore_data_df.head()

App        Category  Rating  \
0     Photo Editor & Candy Camera & Grid & ScrapBook  ART_AND_DESIGN     4.1   
1                                Coloring book moana  ART_AND_DESIGN     3.9   
2  U Launcher Lite – FREE Live Cool Themes, Hide ...  ART_AND_DESIGN     4.7   
3                              Sketch - Draw & Paint  ART_AND_DESIGN     4.5   
4              Pixel Draw - Number Art Coloring Book  ART_AND_DESIGN     4.3   

  Reviews  Size     Installs  Type Price Content Rating  \
0     159   19M      10,000+  Free     0       Everyone   
1     967   14M     500,000+  Free     0       Everyone   
2   87510  8.7M   5,000,000+  Free     0       Everyone   
3  215644   25M  50,000,000+  Free     0           Teen   
4     967  2.8M     100,000+  Free     0       Everyone   

                      Genres      Last Updated         Current Ver  \
0               Art & Design   January 7, 2018               1.0.0   
1  Art & Design;Pretend Play  January 15, 2018               2.0.0   
2               Art & Design    August 1, 2018               1.2.4   
3               Art & Design      June 8, 2018  Varies with device   
4    Art & Design;Creativity     June 20, 2018                 1.1   

    Android Ver  
0  4.0.3 and up  
1  4.0.3 and up  
2  4.0.3 and up  
3    4.2 and up  
4    4.4 and up

In [6]:
user_reviews_df.head()

App                                  Translated_Review  \
0  10 Best Foods for You  I like eat delicious food. That's I'm cooking ...   
1  10 Best Foods for You    This help eating healthy exercise regular basis   
2  10 Best Foods for You                                                NaN   
3  10 Best Foods for You         Works great especially going grocery store   
4  10 Best Foods for You                                       Best idea us   

  Sentiment  Sentiment_Polarity  Sentiment_Subjectivity  
0  Positive                1.00                0.533333  
1  Positive                0.25                0.288462  
2       NaN                 NaN                     NaN  
3  Positive                0.40                0.875000  
4  Positive                1.00                0.300000

In [7]:
playstore_data_df.sample(10)

App         Category  Rating Reviews  \
8056  Theme for Tecno Camon CX / C8        LIFESTYLE     4.6     147   
8956                4K Ultra Camera      PHOTOGRAPHY     4.2     382   
7876                     CT Lottery           FAMILY     3.8     271   
2029    Dog Run - Pet Dog Simulator           FAMILY     4.7   48615   
5255                  AJ and Alyssa        LIFESTYLE     NaN       0   
8922               Clean My Android     PRODUCTIVITY     4.7  101163   
9904                      Brexit EU           FAMILY     4.4       7   
9808           HTC Sense Input - ES            TOOLS     4.3    3509   
1761                  Magic Tiles 3             GAME     4.5  592282   
3449  Lavender Emoji Keyboard Theme  PERSONALIZATION     4.1   12008   

                    Size     Installs  Type Price Content Rating  \
8056                9.8M      10,000+  Free     0       Everyone   
8956                2.9M     100,000+  Free     0       Everyone   
7876                8.3M      50,000+  Free     0           Teen   
2029                 24M  10,000,000+  Free     0       Everyone   
5255                475k         100+  Free     0       Everyone   
8922                6.2M   5,000,000+  Free     0       Everyone   
9904                7.5M         500+  Free     0       Everyone   
9808                6.5M   1,000,000+  Free     0       Everyone   
1761  Varies with device  50,000,000+  Free     0       Everyone   
3449                6.1M   1,000,000+  Free     0       Everyone   

                             Genres      Last Updated Current Ver  \
8056                      Lifestyle       May 2, 2017       1.0.1   
8956                    Photography     July 30, 2018   39.272.12   
7876                  Entertainment     June 12, 2017       1.2.0   
2029  Simulation;Action & Adventure      May 22, 2018      1.4.12   
5255                      Lifestyle      June 1, 2017       1.0.0   
8922                   Productivity     July 31, 2018     1.4.5.2   
9904                         Casual    March 10, 2016         1.0   
9808                          Tools  October 30, 2015  1.0.612919   
1761                          Music    August 3, 2018    5.13.007   
3449                Personalization      May 27, 2016         5.0   

       Android Ver  
8056    4.0 and up  
8956    2.2 and up  
7876    4.0 and up  
2029    4.1 and up  
5255  2.3.3 and up  
8922    4.1 and up  
9904    2.3 and up  
9808    5.0 and up  
1761    4.1 and up  
3449    4.0 and up

In [8]:
print(playstore_data_df.shape)
print(user_reviews_df.shape)

(10841, 13)
(64295, 5)


# **Description of App Dataset columns**#




App : The name of the app

Category : The category of the 

Rating : The rating of the app in the Play Store

Reviews : The number of reviews of the app

Size : The size of the app

Installs : The number of installs of the app

Type : The type of the app (Free/Paid)

price: The price of the app (0 if it is Free)

Content Rating :The appropiate target audience of the app

Genres: The genre of the app

Last Updated : The date when the app was last updated

Current Ver : The current version of the app

Android Ver : The minimum Android version required to run the app

In [9]:
playstore_data_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 10841 entries, 0 to 10840
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             10841 non-null  object 
 1   Category        10841 non-null  object 
 2   Rating          9367 non-null   float64
 3   Reviews         10841 non-null  object 
 4   Size            10841 non-null  object 
 5   Installs        10841 non-null  object 
 6   Type            10840 non-null  object 
 7   Price           10841 non-null  object 
 8   Content Rating  10840 non-null  object 
 9   Genres          10841 non-null  object 
 10  Last Updated    10841 non-null  object 
 11  Current Ver     10833 non-null  object 
 12  Android Ver     10838 non-null  object 
dtypes: float64(1), object(12)
memory usage: 1.1+ MB


By diagnosing the data frame, we know that:

There are 13 columns of properties with 10841 rows of data.

Column 'Reviews', 'Size', 'Installs' and 'Price' are in the type of 'object'

Values of column 'Size' are strings representing size in 'M' as Megabytes, 'k' as kilobytes and also 'Varies with devices'.

Values of column 'Installs' are strings representing install amount with symbols such as '+'.

Values of column 'Price' are strings representing price with symbol '$'.

Hence, we will need to do some data cleaning

## ***.DATA CLEANING***

In [10]:
# finding missing values in the data
print( ' ' ' missing values are as following' + '\n\n' + ' ' '*'*20)

playstore_data_df.isnull().sum()

  missing values are as following

 * * * * * * * * * * * * * * * * * * * *


App                  0
Category             0
Rating            1474
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

From the above result , it can be stated that the rating column has highest amount of null values i.e.,1474 ,current version column has 8 null values, Android version column has 3 null values and content rating and type both have 1 null value each .And the rest has no null values. 



Data can contain duplicate values,NaN type and wrong datatype so that we need to clean the data

In [11]:
#checking for duplicate values 
playstore_data_df[playstore_data_df.duplicated(subset=['App'], keep='first' )]

App       Category  Rating Reviews  \
229          Quick PDF Scanner + OCR FREE       BUSINESS     4.2   80805   
236                                   Box       BUSINESS     4.2  159872   
239                    Google My Business       BUSINESS     4.4   70991   
256                   ZOOM Cloud Meetings       BUSINESS     4.4   31614   
261             join.me - Simple Meetings       BUSINESS     4.0    6989   
...                                   ...            ...     ...     ...   
10715                  FarmersOnly Dating         DATING     3.0    1145   
10720  Firefox Focus: The privacy browser  COMMUNICATION     4.4   36981   
10730                         FP Notebook        MEDICAL     4.5     410   
10753      Slickdeals: Coupons & Shopping       SHOPPING     4.5   33599   
10768                                AAFP        MEDICAL     3.8      63   

                     Size     Installs  Type Price Content Rating  \
229    Varies with device   5,000,000+  Free     0       Everyone   
236    Varies with device  10,000,000+  Free     0       Everyone   
239    Varies with device   5,000,000+  Free     0       Everyone   
256                   37M  10,000,000+  Free     0       Everyone   
261    Varies with device   1,000,000+  Free     0       Everyone   
...                   ...          ...   ...   ...            ...   
10715                1.4M     100,000+  Free     0     Mature 17+   
10720                4.0M   1,000,000+  Free     0       Everyone   
10730                 60M      50,000+  Free     0       Everyone   
10753                 12M   1,000,000+  Free     0       Everyone   
10768                 24M      10,000+  Free     0       Everyone   

              Genres       Last Updated         Current Ver  \
229         Business  February 26, 2018  Varies with device   
236         Business      July 31, 2018  Varies with device   
239         Business      July 24, 2018    2.19.0.204537701   
256         Business      July 20, 2018      4.1.28165.0716   
261         Business      July 16, 2018           4.3.0.508   
...              ...                ...                 ...   
10715         Dating  February 25, 2016                 2.2   
10720  Communication       July 6, 2018                 5.2   
10730        Medical     March 24, 2018           2.1.0.372   
10753       Shopping      July 30, 2018                 3.9   
10768        Medical      June 22, 2018               2.3.1   

              Android Ver  
229          4.0.3 and up  
236    Varies with device  
239            4.4 and up  
256            4.0 and up  
261            4.4 and up  
...                   ...  
10715          4.0 and up  
10720          5.0 and up  
10730          4.4 and up  
10753          4.4 and up  
10768          5.0 and up  

[1181 rows x 13 columns]

The above output retrieved 1181 rows of duplicate data. So we need to drop this duplicate data

In [12]:
#droping the  duplicates from App column.
playstore_data_df.drop_duplicates(subset=['App'],keep='first', inplace=True, ignore_index=True)


Checking if the duplicate data is removed or not.

In [13]:
playstore_data_df[playstore_data_df.duplicated(subset=['App'], keep='first' )]

Empty DataFrame
Columns: [App, Category, Rating, Reviews, Size, Installs, Type, Price, Content Rating, Genres, Last Updated, Current Ver, Android Ver]
Index: []

All the duplicate columns are successfully dropped.

In [14]:
playstore_data_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9660 entries, 0 to 9659
Data columns (total 13 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   App             9660 non-null   object 
 1   Category        9660 non-null   object 
 2   Rating          8197 non-null   float64
 3   Reviews         9660 non-null   object 
 4   Size            9660 non-null   object 
 5   Installs        9660 non-null   object 
 6   Type            9659 non-null   object 
 7   Price           9660 non-null   object 
 8   Content Rating  9659 non-null   object 
 9   Genres          9660 non-null   object 
 10  Last Updated    9660 non-null   object 
 11  Current Ver     9652 non-null   object 
 12  Android Ver     9657 non-null   object 
dtypes: float64(1), object(12)
memory usage: 981.2+ KB


In [15]:
playstore_data_df.isnull().sum()

App                  0
Category             0
Rating            1463
Reviews              0
Size                 0
Installs             0
Type                 1
Price                0
Content Rating       1
Genres               0
Last Updated         0
Current Ver          8
Android Ver          3
dtype: int64

 Here The Rating seem to have a very high number of missing observations.The current ver, Android ver, Type and content rating also have missing values, but they’re pretty negligible compared to Rating.



Replacing all null values with their appropriate values using **aggregate** **functions** such as **mean**,**median**,**mode**



In [16]:
# Replacing the null values in rating column with the median.

median_val =round(playstore_data_df['Rating'].median(),2)
#median_val
playstore_data_df['Rating'].fillna(median_val,inplace=True)

Replacing all null values in 'Current version','Android version','Content Rating'and 'Type'columns with mode


In [17]:
playstore_data_df['Current Ver'].fillna(playstore_data_df['Current Ver'].mode()[0],inplace=True)
playstore_data_df['Android Ver'].fillna(playstore_data_df['Android Ver'].mode()[0],inplace=True)
playstore_data_df['Content Rating'].fillna(playstore_data_df['Content Rating'].mode()[0],inplace = True)
playstore_data_df['Type'].fillna(playstore_data_df['Type'].mode()[0],inplace = True)

In [18]:
playstore_data_df.isnull().sum()


App               0
Category          0
Rating            0
Reviews           0
Size              0
Installs          0
Type              0
Price             0
Content Rating    0
Genres            0
Last Updated      0
Current Ver       0
Android Ver       0
dtype: int64